link/src/test/java/com/stripe/android/link/ui/verification/VerificationViewModelTest.kt

```diff
@@ -142,6 +142,34 @@ class VerificationViewModelTest {
         verify(linkAccountManager).logout()
     }
 
+    @Test
+    fun `Resending code is reflected in state`() = runTest {
+        whenever(linkAccountManager.startVerification()).thenReturn(Result.success(mock()))
+
+        val viewModel = createViewModel().apply {
+            resendCode()
+        }
+
+        val state = viewModel.viewState.value
+        assertThat(state.isSendingNewCode).isFalse()
+        assertThat(state.didSendNewCode).isTrue()
+    }
```

Comment:
I don’t feel great about these tests. Due to `LinkAccountManager` being mocked (as opposed to being a fake implementation), I don’t believe we can achieve a test where we test both the intermediate state (_while_ requesting the code) and the final state (_after_ requesting the code).

Thoughts on separating `LinkAccountManager` into interface and implementation? Would make testing easier.

Reply:
You can mock that behavior with something like this: https://github.com/stripe/stripe-android/blob/811d48d4ba28f59ee2be49bc122d1df5d19de6e0/link/src/test/java/com/stripe/android/link/ui/signup/SignUpViewModelTest.kt#L129-L135

Reply:
Oh interesting, I didn’t know about that. I’ll go ahead with that for now, but might pick up the refactor later as a separate pull request.